pkg/storagev2/leveldbstore/batch.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package leveldbstore
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	ldb "github.com/syndtr/goleveldb/leveldb"
+)
+
+func (s *Store) Batch(ctx context.Context) (storage.Batch, error) {
+	return &Batch{
+		ctx:   ctx,
+		mu:    new(sync.Mutex),
+		batch: new(ldb.Batch),
+		store: s,
+	}, nil
+}
+
+type Batch struct {
+	mu  *sync.Mutex
```

Comment:
I dont think this also needs to be a pointer.

Reply:
fixed